mozetl/taar/taar_amowhitelist.py

```diff
@@ -0,0 +1,86 @@
+#!/bin/env python
+
+import click
+import json
+import logging
+import logging.config
+from dateutil.parser import parse
+from tempfile import NamedTemporaryFile
+import datetime
+
+from taar_utils import read_from_s3, store_json_to_s3
+
+AMO_DUMP_BUCKET = 'telemetry-parquet'
+AMO_DUMP_PREFIX = 'telemetry-ml/addon_recommender/'
+
+AMO_DUMP_BASE_FILENAME = 'extended_addons_database'
+FILTERED_AMO_BASE_FILENAME = 'whitelist_addons_database'
+
+AMO_DUMP_FILENAME = AMO_DUMP_BASE_FILENAME + '.json'
+FILTERED_AMO_FILENAME = FILTERED_AMO_BASE_FILENAME + '.json'
+
+MIN_RATING = 3.0
+MIN_AGE = 60
+
+logger = logging.getLogger('amo_whitelist')
+
+
+class AMOTransformer:
+    def __init__(self, bucket, prefix, fname, min_rating, min_age):
+        self._s3_bucket = bucket
+        self._s3_prefix = prefix
+        self._s3_fname = fname
+        self._min_rating = min_rating
+        self._min_age = min_age
+
+    def extract(self):
+        with NamedTemporaryFile() as dst_json_file:
+            read_from_s3(dst_json_file.name,
+                         self._s3_fname,
+                         self._s3_prefix,
+                         self._s3_bucket)
+
+            data = open(dst_json_file.name).read()
+            return json.loads(data)
+
+    def transform(self, json_data):
```

Comment:
Let's document this: what are the criteria here? Why are we using them? Was this discussed somewhere that can be referenced from the code (e.g. link?)?

Reply:
Comments and links to the issue added now.